craft_parts/packages/snaps.py

```diff
@@ -339,12 +339,12 @@ def _get_parsed_snap(snap: str) -> Tuple[str, str]:
     return snap_name, snap_channel
 
 
-def get_snapd_socket_path_template():
+def get_snapd_socket_path_template() -> str:
     """Return the template for the snapd socket URI."""
     return "http+unix://%2Frun%2Fsnapd.socket/v2/{}"
 
 
-def _get_local_snap_file_iter(snap_name: str, *, chunk_size: int):
+def _get_local_snap_file_iter(snap_name: str, *, chunk_size: int) -> Iterator:
```

Comment:
The [typeshed](https://github.com/python/typeshed/blob/main/stubs/requests/requests/models.pyi#L109) for requests agrees with what's written here, but I'm pretty sure the `Response` object's `iter_content` will return either an `Iterator[str]` if `decode_unicode=True` or an `Iterator[bytes]` if `decode_unicode=False`.

(Likely relevant typeshed issue: https://github.com/python/typeshed/issues/9550)

Reply:
Thanks, update to `Iterator[bytes]`
good call